spec/features/consumer/registration_spec.rb

```diff
@@ -94,6 +97,43 @@
       expect(e.twitter).to eq "@TwItTeR"
       expect(e.instagram).to eq "@InStAgRaM"
     end
+
+  end
+
+  describe "Terms of Service agreement" do
+    let!(:user2) { create(:user) }
+
+    before do
+      quick_login_as user2
+    end
+
+    context "if accepting Terms of Service is not required" do
+      before { Spree::Config.enterprises_require_tos = false }
+
+      it "allows registration as normal" do
+        visit registration_path
+
+        click_button "Let's get started!"
+        find("div#progress-bar").visible?.should be_true
+      end
+    end
+
+    context "if accepting Terms of Service is required" do
+      before { Spree::Config.enterprises_require_tos = true }
+
+      it "does not allow registration unless checkbox is checked" do
+        visit registration_path
+
+        page.should have_content('Terms of Service')
+        page.should have_selector("input.button.primary[disabled]")
+
```

Comment:
@oeoeaio @RohanM I'm having serious trouble with Spree::Config values having unpredictable values in tests... 

Before the test runs, `Spree::Config.enterprises_require_tos` is set to true. The page displays different content depending on this setting, determined by the view which renders an enclosing element with the attribute `ng-init = "tos_required=#{Spree::Config.enterprises_require_tos}" ` and hides elements based on this value.

The tests here fail though, with strange results, as if the config value has not been changed. After `visit_registration path:`
 If I add in `expect(Spree::Config.enterprises_require_tos).to eq true` it passes.
 If I add in `html.should include('tos_required=true')` it fails.
 If I add in `html.should_not include('tos_required=false')` it fails.

So it's confirming that when the test runs and after the page is loaded, `Spree::Config.enterprises_require_tos` is set to true, but `#{Spree::Config.enterprises_require_tos}` in being rendered as false in the DOM at the same time..?

This behaviour is only present in tests, not in the dev environment. Also, the results seem to change unexpectedly, sometimes passing and sometimes failing. There are 3 tests in this file where the content is altered by the config setting. Sometimes the value seems to be false in all tests, sometimes true in all tests, and sometimes set to false or true as expected (it's explicitly set before each test as true or false). For added weirdness: changing the value in the browser in the dev environment seems to effect the outcome in the tests...?

Am I going mad or missing something really obvious?